dask/dataframe/tests/test_shuffle.py

```diff
@@ -1182,6 +1182,17 @@ def test_set_index_overlap_2():
     assert ddf2.npartitions == 8
 
 
+def test_set_index_overlap_3():
```

Comment:
I see that we have `test_set_index_overlap` and then `_2` and  `_3`. I thought the only thing changing was the number of partitions but it seems it's not just that. 

Would it be better to have either a comment/docstring to these tests that explains better what each test is accomplishing?

Reply:
yeah it's a lazy name, I did link out to the issue that has the full context. But I'm happy to change the name